xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -822,47 +1452,180 @@ CGUIFontTTF::Character* CGUIFontTTF::GetCharacter(character_t chr, FT_UInt glyph
   m_nestedBeginCount = nestedBeginCount;
 
   // update the lookup table with only the m_char addresses that have changed
-  for (size_t i = startIndex; i < m_char.size(); ++i)
+  for (size_t i = startIndex; i < fontData->m_char.size(); ++i)
   {
-    if (m_char[i].m_glyphIndex < MAX_GLYPH_IDX)
+    if (fontData->m_char[i].m_glyphIndex < MAX_GLYPH_IDX)
     {
-      // >> 16 is style (0-6), then 16 - 12 (>> 4) is equivalent to style * 4096
-      character_t ch = ((m_char[i].m_glyphAndStyle & 0xffff0000) >> 4) | m_char[i].m_glyphIndex;
+      // >> 32 is style (0-6), then 32 - 12 (>> 20) is equivalent to style * 4096
+      const uint32_t ch = ((fontData->m_char[i].m_glyphAndStyle & 0xFFFFFFFF00000000) >> 20) |
+                          fontData->m_char[i].m_glyphIndex;
 
-      if (ch < LOOKUPTABLE_SIZE)
-        m_charquick[ch] = m_char.data() + i;
+      if (ch < LOOKUPTABLE_SIZE && fontData->m_mainFont)
```

Comment:
If I'm not wrong `m_charquick` is only used with main font. If so, it would be better not execute this for loop at all (if font is not main)

```c++
if (fontData->m_mainFont)
{
  // update the lookup table with only the m_char addresses that have changed
  for (...
}
```

Reply:
You are right, yes to make the `if` before is much better and changed it.